src/api/services.py

```
@@ -102,4 +88,93 @@ def update_spot(spot, validated_data):
+            logger.warning(f"Skipping cancellation confirmation for {booking.id}. User email is missing.")
+            return False
+
+        context = {'booking': booking, 'user': booking.user}
+        
+        try:
+            lot_name = getattr(getattr(booking.spot, 'lot', None), 'name', 'Unknown Lot')
+            subject = f" Booking Cancelled: #{booking.id} | {lot_name}"
+            message_html = render_to_string('emails/booking_cancellation.html', context)
+            message_text = render_to_string('emails/booking_cancellation_text.txt', context)
+            
+        except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)